src/provider/provider_os_memory_posix.c

```diff
@@ -106,3 +109,48 @@ void os_strerror(int errnum, char *buf, size_t buflen) {
         LOG_PERR("Retrieving error code description failed");
     }
 }
+
+// open a file
+int os_file_open(const char *path) {
+    if (path == NULL) {
+        LOG_ERR("empty path");
+        return -1;
+    }
+
+    int fd = open(path, O_RDWR);
+    if (fd == -1) {
+        LOG_PERR("cannot open the file: %s", path);
```

Comment:
pls add negative tests for these

Reply:
Hmm ...

Reply:
Done

Reply:
hmm.. what about empty path negative test?

Reply:
Added.
Done